In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# Any results you write to the current directory are saved as output.
import warnings
warnings.filterwarnings('ignore')

/kaggle/input/titanic/titanic_cleaned.csv


Now Let's load the dataset

In [2]:
df=pd.read_csv('/kaggle/input/titanic/titanic_cleaned.csv')
df.head()

Survived  Pclass  Sex   Age     Fare  Family_count  Cabin_ind
0         0       3    0  22.0   7.2500             1          0
1         1       1    1  38.0  71.2833             1          1
2         1       3    1  26.0   7.9250             0          0
3         1       1    1  35.0  53.1000             1          1
4         0       3    0  35.0   8.0500             0          0

Let's explore the dataset a bit.

In [3]:
df.shape

(891, 7)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 7 columns):
Survived        891 non-null int64
Pclass          891 non-null int64
Sex             891 non-null int64
Age             891 non-null float64
Fare            891 non-null float64
Family_count    891 non-null int64
Cabin_ind       891 non-null int64
dtypes: float64(2), int64(5)
memory usage: 48.9 KB


In [5]:
df.isnull().sum()

Survived        0
Pclass          0
Sex             0
Age             0
Fare            0
Family_count    0
Cabin_ind       0
dtype: int64

Now let's separate the target column from feature column

In [6]:
X=df.drop(columns=['Survived'])
y=df[['Survived']]
print(X.shape)
print(y.shape)

(891, 6)
(891, 1)


Now split the dataset into train and test by 80:20

In [7]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test=train_test_split(X, y, test_size=0.2, random_state=42)
print(X_train.shape)
print(y_test.shape)

(712, 6)
(179, 1)


1. Now Let's train the models-in fact 6 models

In [8]:
from sklearn.linear_model import LogisticRegression, SGDClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.naive_bayes import GaussianNB

lr=LogisticRegression().fit(X_train, y_train)
sgd=SGDClassifier().fit(X_train, y_train)
rf=RandomForestClassifier().fit(X_train, y_train)
dtc=DecisionTreeClassifier().fit(X_train, y_train)
svc=SVC().fit(X_train, y_train)
nb=GaussianNB().fit(X_train, y_train)

y_lr=lr.predict(X_test)
y_sgd=sgd.predict(X_test)
y_rf=rf.predict(X_test)
y_dtc=dtc.predict(X_test)
y_svc=svc.predict(X_test)
y_nb=nb.predict(X_test)

Now evaluate the model using classification report from scikit-learn

In [9]:
from sklearn.metrics import classification_report
print('Logistic Regression: ',classification_report(y_lr, y_test) )
print('Stochastic Gradient Descent: ',classification_report(y_sgd, y_test) )
print('Random Forest: ',classification_report(y_rf, y_test) )
print('Decision Tree: ',classification_report(y_dtc, y_test) )
print('Support Vector: ',classification_report(y_svc, y_test) )
print('Naive Bayes: ',classification_report(y_nb, y_test) )

Logistic Regression:                precision    recall  f1-score   support

           0       0.88      0.83      0.85       111
           1       0.74      0.81      0.77        68

    accuracy                           0.82       179
   macro avg       0.81      0.82      0.81       179
weighted avg       0.83      0.82      0.82       179

Stochastic Gradient Descent:                precision    recall  f1-score   support

           0       0.67      0.90      0.77        78
           1       0.89      0.65      0.75       101

    accuracy                           0.76       179
   macro avg       0.78      0.78      0.76       179
weighted avg       0.79      0.76      0.76       179

Random Forest:                precision    recall  f1-score   support

           0       0.88      0.81      0.84       113
           1       0.72      0.80      0.76        66

    accuracy                           0.81       179
   macro avg       0.80      0.81      0.80       179
weight

Observation: Logistic Regression is the best classifier with accuracy rate 82%

Great job! well done.